lib/handlers/connect.js

```diff
@@ -187,13 +187,16 @@ function storeWill (arg, done) {
   var client = this.client
   client.will = client._will
   if (client.will) {
-    client.broker.persistence.putWill(
-      client,
-      client.will,
-      done)
-    return
+    // delete any existing will messages from persistence
+    client.broker.persistence.delWill(client, function () {
```

Comment:
I think this should be deleted nevertheless.

Reply:
Sure